File: zeno/classes/report.py
from enum import Enum
from typing import List, Union

from zeno.classes.base import CamelModel


class ChartType(str, Enum):
    BAR = "BAR"
    LINE = "LINE"
    TABLE = "TABLE"
    BEESWARM = "BEESWARM"
    RADAR = "RADAR"
    HEATMAP = "HEATMAP"


class Parameters(CamelModel):
    x_encoding: str
    y_encoding: str
    z_encoding: str
    fixed_dimension: str
    second_slices: List[str]


class Report(CamelModel):
    name: str
    type: Union[ChartType, None] = None
    slices: Union[List[str], None] = None
    metrics: Union[List[str], None] = None
    models: Union[List[str], None] = None
    parameters: Union[Parameters, None] = None

File: frontend/src/zenoservice/models/Report.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ChartType } from "./ChartType";
import type { Parameters } from "./Parameters";

export type Report = {
	name: string;
	type?: ChartType;
	slices?: Array<string>;
	metrics?: Array<string>;
	models?: Array<string>;
	parameters?: Parameters;
};
